### ISSUE TITLE
Failed to create fix for UnnecessaryMap: Should have two arguments

### ISSUE THREAD

**Issue by qarmin** [astral-sh/ruff#6951]
Ruff 0.0.286 (latest changes from main branch)

```
ruff  *.py --select ALL --no-cache
```

file content:
```
def ists2dict(keys, values):
   return dict(map(lambda k, v: (k, v), keys, values))
```

error:
```
error: Failed to create fix for UnnecessaryMap: Should have two arguments
```

[_helpers-42522735449234892253272 (10th copy)044.py.zip](https://github.com/astral-sh/ruff/files/12457792/_helpers-42522735449234892253272.10th.copy.044.py.zip)

**Comment by zanieb** [comment]
Bails at [Should have two arguments](https://github.com/astral-sh/ruff/blob/fc89976c24d1c8b9d13d88ef70b619e8a2ac393a/crates/ruff/src/rules/flake8_comprehensions/fixes.rs#L1014) 

The following is valid code at runtime:
```python
map(lambda x, y: x + y, [1], [2])
```

It seems like the fix would need to be updated to zip the arguments when there are multiple.
